### ISSUE TITLE
RESMON very high

### ISSUE THREAD

**Issue by Legacy-TacticalGamingInteractive** [traditionalism/geneva-vending#1]
![image](https://user-images.githubusercontent.com/25211271/230697349-71c28204-0007-41a9-a844-0bb366ac7c39.png)

Would anyone know how we could possibly improve this?

**Comment by traditionalism** [comment]
Usage for me only goes up as high as 0.7ms (while I'm standing right in front of the vending-machine, being prompted), otherwise, it's around 0.4ms. When exactly are you doing when this picture is taken? I have plans to optimize the resource.

**Comment by Legacy-TacticalGamingInteractive** [comment]
> Usage for me only goes up as high as 0.7ms (while I'm standing right in front of the vending-machine, being prompted), otherwise, it's around 0.4ms. When exactly are you doing when this picture is taken? I have plans to optimize the resource.

This is server side RESMON
60+ Players

**Comment by traditionalism** [comment]
I'm afraid there is nothing I can do about this. I've looked into it, and even with no code running in the server script it still has very high usage. I've looked at a few of my other C# resources, and some other people's C# resources and it seems like the same deal. Although the usage shouldn't be a problem to be quite honest with you, I'll be pushing some optimizations soon here either way. I'll keep this issue open just in case anything is figured out.

**Comment by Legacy-TacticalGamingInteractive** [comment]
I have heard that one of the major issues is C# in of itself being not so optimized compared to lua scripts when it comes to FiveM but who knows.

Whats also weird is that I have ox_inventory which is supposed to come with vending machines that work, only they don't for me and so therefore I've needed to use your script.

**Comment by traditionalism** [comment]
Yeah, C# does have a bit more overhead compared to Lua. But, I like it a bit more, so I decided to use it with this resource. I made this just for fun, and a more accurate replica of the online vending machines compared to the ones I've seen.
